cloud-storage.go

```diff
@@ -123,7 +132,25 @@ func newListIterator(f func() (*ListObject, error)) *ListIterator {
 func mergeOpts(opts ...*CloudStorageOption) *CloudStorageOption {
```

Comment:
can you please show me a user scenario that there're multiple options exist, why the consumer put into 1 struct, is there anything I missing?

Reply:
ah, sorry, previously, multiple opts to prevent breaking changes in `NewCloudStorage`, bcs by using `...*CloudStorageOption` its not mandatory param and could prevent breaking changes in method `NewCloudStorage`. 

Since we end up with new method, i updated using `CloudStorage` as param